platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/ReadableRosterStoreImpl.java

```diff
@@ -128,4 +131,42 @@ public Bytes getPreviousRosterHash() {
                 .filter(pair -> rosterMap.contains(new ProtoBytes(pair.activeRosterHash())))
                 .toList();
     }
+
+    /**
+     * @return the node ids from the roster entries of both active and candidate rosters combined
+     */
+    public List<EntityNumber> getCombinedRosterEntriesNodeIds() {
+        final var activeRoster = getActiveRoster();
+        final var candidateRoster = getCandidateRoster();
+        List<EntityNumber> nodeEntriesInRosters = List.of();
+
+        if (activeRoster != null && candidateRoster != null) {
+            final Stream<Long> activeRosterEntries =
+                    activeRoster.rosterEntries().stream().map(RosterEntry::nodeId);
+            final Stream<Long> candidateRosterEntries =
+                    candidateRoster.rosterEntries().stream().map(RosterEntry::nodeId);
+            nodeEntriesInRosters = Stream.concat(activeRosterEntries, candidateRosterEntries)
+                    .distinct()
+                    .map(EntityNumber::new)
+                    .toList();
+        }
+        // If we are here, at least one of the rosters is null
```

Comment:
There isn't an `else` or a return statement, so we will always get `here`.   Should a return be added or an else added to make sure that the following cases are skipped? 

Would it be simpler and collapse all cases to do something like this: 

```
final Stream<Long> activeRosterEntries = activeRoster == null ? Stream.empty() : 
                    activeRoster.rosterEntries().stream().map(RosterEntry::nodeId);
final Stream<Long> candidateRosterEntries = candidateRoster == null ? Stream.empty() : 
                    candidateRoster.rosterEntries().stream().map(RosterEntry::nodeId);
return Stream.concat(activeRosterEntries, candidateRosterEntries)
                    .distinct()
                    .map(EntityNumber::new)
                    .toList();
```

Reply:
Yes, I agree, this simplifies the logic a lot, ty!